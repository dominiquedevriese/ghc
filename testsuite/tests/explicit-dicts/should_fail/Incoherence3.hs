{-# LANGUAGE TypeFamilies #-}
module Incoherence3 where

import GHC.Classes (Eq(..))

foo :: (Eq a, a ~ b, Eq b) => a -> b -> Bool
foo x y = x == y

bar :: Eq q => Eq.Dict q -> q -> q -> Bool
bar dict = foo ((dict :: Eq a))
