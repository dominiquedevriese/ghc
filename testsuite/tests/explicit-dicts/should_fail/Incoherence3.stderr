
Incoherence3.hs:10:12: error: [-Wincoherence (in -Wdefault), -Werror=incoherence]
    • Explicit dictionary application to: ‘Eq a’
      is not allowed because of possible incoherence:
        ‘Eq a’ is entailed by the remainder of the context: ‘(a ~ b, Eq b)’
    • In the expression: foo dict
      In an equation for ‘bar’: bar dict = foo dict
